tests/PathResolverTest.php

```
@@ -0,0 +1,34 @@
+        // In development (non-PHAR), should find the local asd-sync/dot.js
+        $dotJsPath = PathResolver::getDotJsPath();
+        
+        // Should be a valid path ending with dot.js
+        $this->assertStringEndsWith('/asd-sync/dot.js', $dotJsPath);
+        $this->assertFileExists($dotJsPath);
+    }
+
+    public function testPathResolverHasProperErrorHandling(): void
+    {
+        // Verify the method exists and has proper error handling capability
+        $this->assertTrue(method_exists(PathResolver::class, 'getDotJsPath'));
```

PHPStan (2.1.17)

Call to function method_exists() with 'Koriym\\AppStateDiagram\\PathResolver' and 'getDotJsPath' will always evaluate to true.


(function.alreadyNarrowedType)